### PR TITLE
{Quota}Generate quota with codegen

### PR DESCRIPTION
**Pull request by ZengTaoxu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Quota

File: src/quota/setup.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+with open('HISTORY.rst', 'r', encoding='utf-8') as f:
+    HISTORY = f.read()
+
+setup(
+    name='quota',
+    version=VERSION,
+    description='Microsoft Azure Command-Line Tools AzureQuotaExtensionAPI Extension',
+    author='Microsoft Corporation',
+    author_email='azpycli@microsoft.com',
+    url='https://github.com/Azure/azure-cli-extensions/tree/master/src/quota',
```
Comment (necusjz) [inline src/quota/setup.py:51]:
```suggestion
    url='https://github.com/Azure/azure-cli-extensions/tree/main/src/quota',
```

Comment (ZengTaoxu) [reply src/quota/setup.py:51]: Thank you!

File: src/quota/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+##### List #####
+```
+az quota usage list \
+    --scope "subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Network/locations/eastus" 
+```
+##### List #####
+```
+az quota usage list \
+    --scope "subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.MachineLearningServices/locations/eastus" 
+```
```
Comment (kairu-ms) [inline src/quota/README.md:26]: Why the help is duplicated?

Comment (ZengTaoxu) [reply src/quota/README.md]: It has been modified

File: src/quota/azext_quota/generated/_help.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+"""
+
+helps['quota create'] = """
+    type: command
+    short-summary: "Create the quota limit for the specified resource with the requested value. To update the quota, \
+follow these steps: 1. Use the GET operation for quotas and usages to determine how much quota remains for the \
+specific resource and to calculate the new quota limit. These steps are detailed in [this \
+example](https://techcommunity.microsoft.com/t5/azure-governance-and-management/using-the-new-quota-rest-api/ba-p/21836\
+70). 2. Use this PUT operation to update the quota limit. Please check the URI in location header for the detailed \
+status of the request."
```
Comment (kairu-ms) [inline src/quota/azext_quota/generated/_help.py:98]: Please overwrite this help in manual folder. The short-summary is too long.

```suggestion
    short-summary: "Create the quota limit for the specified resource with the requested value."
```

File: src/quota/azext_quota/generated/_help.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+--limit-object value=10 limit-object-type=LimitValue --resource-type lowPriority
+"""
+
+helps['quota update'] = """
+    type: command
+    short-summary: "Update the quota limit for a specific resource to the specified value: 1. Use the Usages-GET and \
+Quota-GET operations to determine the remaining quota for the specific resource and to calculate the new quota limit. \
+These steps are detailed in [this example](https://techcommunity.microsoft.com/t5/azure-governance-and-management/using\
+-the-new-quota-rest-api/ba-p/2183670). 2. Use this PUT operation to update the quota limit. Please check the URI in \
+location header for the detailed status of the request."
```
Comment (kairu-ms) [inline src/quota/azext_quota/generated/_help.py:136]:
```suggestion
    short-summary: "Update the quota limit for a specific resource to the specified value."
```

**Review by kairu-ms — COMMENTED** [review]
The commands looks good to me. Please update the short-summery of 'quota create' and 'quota update'.

File: src/quota/azext_quota/generated/_help.py (hunk 3 above)
Comment (ZengTaoxu) [reply src/quota/azext_quota/generated/_help.py]: It has been modified.

File: src/quota/azext_quota/generated/_help.py (hunk 4 above)
Comment (ZengTaoxu) [reply src/quota/azext_quota/generated/_help.py]: It has been modified.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**